PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m

```diff
@@ -119,7 +119,13 @@ + (NSURLSessionTask *) communicate_with_server:(NSString*)url
             NSLog(@"Error %@", error);
             errorHandlerBlock(response, error);
         } else {
-            completionHandlerBlock(response,[self parse_json:datastring]);
+            @try {
+                completionHandlerBlock(response,[self parse_json:datastring]);
+            }
+            @catch ( NSException *e ) {
+                NSLog(@"Error %@", e);
```

Comment:
what does the actual error look like that is logged here? is it intelligible enough to know what happened?

Reply:
04/05/2018 15:33:23.479 PlutoHelperAgent[76378]: Error *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]

Here is an example above.

Debatable. The error is saying that a nil object is getting passed in as part of a dictionary because of the blank password field.

Reply:
If you like I could change 'Error' to 'Error (It is likely that one or more blank fields where processed)'.

Reply:
@Dala0 could you catch the error closer to where it happens? then you could check which fields that you're expecting don't exist in the dictionary

Reply:
I doubt it. I went through the code line by line trying to find where is breaks and that is as close as I could get in the programme code. Getting any closer would require rewriting external dependencies, as far as I can work out. @fredex42